hedgehog/src/Hedgehog/Internal/Gen.hs

```diff
@@ -1146,8 +1146,8 @@ constant =
 --
 --   /The input list must be non-empty./
 --
-element :: MonadGen m => [a] -> m a
-element = \case
+element :: (Foldable f, MonadGen m) => f a -> m a
+element fa = case toList fa of
```

Comment:
This could also use `null :: t a -> [Bool]` from Data.Foldable, the difference is probably negligible. 

The comment here should be "Hedgehog.Gen.element: used with empty foldable"

Reply:
Thanks, I've updated the error message. I'm not sure `null` would help much - we still need a list in order to use `!!` - so I've left that as is.